### PR TITLE
Sm client

### PR DESCRIPTION
**Pull request by Eabramov84**

### CODE REVIEW

File: proto/servicemanager/v3/servicemanager.proto [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+/***********************************************************************************************************************
+ * SMService
+ **********************************************************************************************************************/
+
+service SMService {
+    rpc RegisterSM(stream SMNotifications) returns (stream CMRequests) {}
+}
+
+message CMRequests {
```
Comment (al1img) [inline proto/servicemanager/v3/servicemanager.proto:15]: SMRequests

File: proto/servicemanager/v3/servicemanager.proto [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    oneof CMRequest {
```
Comment (al1img) [inline proto/servicemanager/v3/servicemanager.proto:16]: ditto

File: proto/servicemanager/v3/servicemanager.proto [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+import "google/protobuf/timestamp.proto";
+
+package servicemanager.v3;
+
+/***********************************************************************************************************************
+ * SMService
+ **********************************************************************************************************************/
+
+service SMService {
+    rpc RegisterSM(stream SMNotifications) returns (stream CMRequests) {}
```
Comment (al1img) [inline proto/servicemanager/v3/servicemanager.proto:12]: Having requests -> responses would be more logical. Let's rename `SMNotifications` -> `SMResponses`

File: proto/servicemanager/v3/servicemanager.proto [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    repeated ServiceInfo services   = 1;
+    repeated LayerInfo layers       = 2;
+    repeated InstanceInfo instances = 3; 
+}
+
+message ServiceInfo {
+    string url            = 1;
+    string service_id     = 2;
+    string provider_id    = 3;
+    uint64 aos_version    = 4;
```
Comment (al1img) [inline proto/servicemanager/v3/servicemanager.proto:50]: Move aos_version, vendor_version, and description into separate struct `VersionInfo`.

File: proto/servicemanager/v3/servicemanager.proto [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+        OverrideEnvVarStatus override_env_var_status  = 7;
+        LogData log                                   = 8;
+        Monitoring monitoring                         = 9;
+        Alert alert                                   = 10;
+    }
+}
+
+message NodeConfiguration {
+   string  node_id                 = 1;
+   repeated string runner_features = 2;
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:139]: string

File: proto/servicemanager/v3/servicemanager.proto (hunk 1 above)
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:15]: renamed to SMIncomingMessages

File: proto/servicemanager/v3/servicemanager.proto (hunk 2 above)
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:16]: SMIncomingMessages

File: proto/servicemanager/v3/servicemanager.proto (hunk 3 above)
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:12]: renamed to SMOutgoingMessages

File: proto/servicemanager/v3/servicemanager.proto (hunk 4 above)
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:50]: done

File: proto/servicemanager/v3/servicemanager.proto [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    string vendor_version = 4;
+    string digest         = 5;
+    string description    = 6;
+    bytes sha256          = 7;
+    bytes sha512          = 8;
+    uint64 size           = 9;
+}
+
+message InstanceInfo {
+    string service_id    = 1;
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:71]: done

File: proto/servicemanager/v3/servicemanager.proto [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    string subject_id    = 2;
+    uint64 num_instances = 3;
+}
+
+message InstanceStateAcceptance {
+    InstanceIdent instance = 1;
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:77]: decided to use instance ident

File: proto/servicemanager/v3/servicemanager.proto [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+        Monitoring monitoring                         = 9;
+        Alert alert                                   = 10;
+    }
+}
+
+message NodeConfiguration {
+   string  node_id                 = 1;
+   repeated string runner_features = 2;
+   uint64 num_cpus                 = 3;
+   uint32 total_ram                = 4;
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:141]: thanks, fixed

File: proto/servicemanager/v3/servicemanager.proto [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+   repeated Partition partitions   = 5;
+}
+
+message Partition{
+    string name          = 1;
+    repeated string type = 2;
```
[the remark this answers is not in the document]
Comment (Eabramov84) [reply proto/servicemanager/v3/servicemanager.proto:147]: to avoid protocol changes decided to use string

**Review by MykolaSolyanko — APPROVED** [review]

File: proto/communicationmanager/v2/updatescheduler.proto [hunk 10]
```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
```
Comment (al1img) [inline proto/communicationmanager/v2/updatescheduler.proto:3]: Fix commit message: PAI -> API?

File: proto/communicationmanager/v2/updatescheduler.proto [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+message SchedulerNotifications {
+    oneof SchedulerNotification {
+        UpdateSOTAStatus sota_status = 1;
+        UpdateFOTAStatus fota_status = 2;
+    }
+}
+
+message UpdateFOTAStatus {
+    UpdateState state = 1;
+    repeated ComponentInfo components = 2;
```
Comment (al1img) [inline proto/communicationmanager/v2/updatescheduler.proto:22]: Could you align field values?

File: proto/communicationmanager/v2/updatescheduler.proto [hunk 12]
```diff
@@ -20,7 +20,7 @@ message SchedulerNotifications {
 message UpdateFOTAStatus {
     UpdateState state = 1;
     repeated ComponentInfo components = 2;
-    BoardConfigInfo board_config = 3;
+    UnitConfigInfo unit_config = 3;
```
Comment (al1img) [inline proto/communicationmanager/v2/updatescheduler.proto:23]: Could you align field values?

**Review by al1img — APPROVED** [review]

**Pull request merged**